### ISSUE TITLE
Unable to determine the workspace. - Error after update to Rider 2018.2

### ISSUE THREAD

**Issue by dillius** [microsoft/azure-devops-intellij#173]
Can no longer retrieve latest from TFS after updating Rider to 2018.2. Worked fine prior, during the same day. On latest 1.1.150 version of the plugin. Not sure if the VSO plugin needs an update or if this is a bug caused by Jetbrains.

> Error:An argument error occurred: Unable to determine the workspace. You may be able to correct this by running 'tf workspaces /collection:TeamProjectCollectionUrl'.

Log details show:

```
16:38:32.141 | INFO  | TFSUpdateEnvironment           | Update on files initiated...
16:38:32.141 | INFO  | VcsHelper                      | getRepositoryContext: cache hit: C:/RiderTFS/Source
16:38:32.141 | INFO  | TFSVcs                         | TFSVcs.getServerContext repositoryContext is null: false
16:38:32.141 | INFO  | ToolRunnerCache                | getRunningToolRunner: toolLocation={0}
16:38:32.142 | INFO  | ToolRunnerCache                | getRunningToolRunner: fast version - 
14.123.1.201709051449
16:38:32.142 | INFO  | ToolRunnerCache                | getRunningToolRunner: key=c:\tf\tf.cmd|
16:38:32.142 | INFO  | ToolRunner                     | sendArgsViaStandardInput: proceedWithArgs: get 
-noprompt C:/RiderTFS/Source C:/Users/mhenderson/.Rider2018.2/config/scratches C:/Users/mhenderson/.Rider2018.2/system/extResources -recursive
16:38:32.142 | INFO  | ToolRunnerCache                | updateCachedInstance: caching a new runner: 
key=c:\tf\tf.cmd|
16:38:32.142 | INFO  | ToolRunner                     | ToolRunner.start: toolLocation = C:\tf\tf.cmd
16:38:32.142 | INFO  | ToolRunner                     | ToolRunner.start: workingDirectory = null
16:38:32.142 | INFO  | ToolRunner                     | arguments: @
16:38:32.319 | INFO  | Command                        | ERROR: An argument error occurred: Unable to 
determine the workspace. You may be able to correct this by running 'tf workspaces /collection:TeamProjectCollectionUrl'.
16:38:32.321 | INFO  | Command                        | CMD: Team Explorer Everywhere Command Line Client 
(version 14.123.1.201709051449)
16:38:32.321 | INFO  | Command                        | CMD: 
16:38:32.322 | INFO  | Command                        | CMD:  get command:
16:38:32.322 | INFO  | Command                        | CMD: 
16:38:32.323 | INFO  | Command                        | CMD:   Retrieves a read-only copy of one or more files 
from the source control
16:38:32.323 | INFO  | Command                        | CMD:   server to the local disk. Any intermediate folders 
are created if necessary. 
16:38:32.323 | INFO  | Command                        | CMD:   
16:38:32.323 | INFO  | Command                        | CMD:   The version to retrieve may be specified 
through the 'version' option or as
16:38:32.323 | INFO  | Command                        | CMD:   a version specification suffix to the item 
specification (example:
16:38:32.323 | INFO  | Command                        | CMD:   '$/file.txt;C34').
16:38:32.323 | INFO  | Command                        | CMD: 
16:38:32.323 | INFO  | Command                        | CMD:  Valid option sets:
16:38:32.323 | INFO  | Command                        | CMD: 
16:38:32.341 | INFO  | Command                        | CMD:   get [/version:<value>] [/recursive] [/preview] 
[/force] [/all] [/overwrite]
16:38:32.341 | INFO  | Command                        | CMD:       [/noautoresolve] [<itemSpec>...]
16:38:32.341 | INFO  | Command                        | CMD: 
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command: Team 
Explorer Everywhere Command Line Client (version 14.123.1.201709051449)
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:  get 
command:
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   
Retrieves a read-only copy of one or more files from the source control
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   server 
to the local disk. Any intermediate folders are created if necessary. 
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   The 
version to retrieve may be specified through the 'version' option or as
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   a 
version specification suffix to the item specification (example:
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   
'$/file.txt;C34').
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:  Valid 
option sets:
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:   get 
[/version:<value>] [/recursive] [/preview] [/force] [/all] [/overwrite]
16:38:32.385 | WARN  | SyncCommand                    | Unknown response from 'tf get' command:       
[/noautoresolve] [<itemSpec>...]
16:38:32.385 | INFO  | Command                        | 243662200(ns) - elapsed time for get -noprompt 
C:/RiderTFS/Source C:/Users/mhenderson/.Rider2018.2/config/scratches C:/Users/mhenderson/.Rider2018.2/system/extResources -recursive
16:38:32.385 | WARN  | RiderApplicationLoadListener   | [STDERR] : 243662200(ns) - elapsed time for get 
-noprompt C:/RiderTFS/Source C:/Users/mhenderson/.Rider2018.2/config/scratches C:/Users/mhenderson/.Rider2018.2/system/extResources -recursive
16:38:32.387 | INFO  | AbstractCommonUpdateAction     | Calling refresh files after update for roots: 
[C:/RiderTFS/Source/, C:/Users/mhenderson/.Rider2018.2/config/scratches/, C:/Users/mhenderson/.Rider2018.2/system/extResources/]
```

**Comment by dillius** [comment]
It seems with the size of the repository I am dealing with the java process started up to try to take care of this gets stuck in some infinite loop; at this point I have simply had to disable the plugin entirely for Rider to be usable again.

**Comment by rbmanian75** [comment]
I am also getting this error on rider.

**Comment by ForNeVeR** [comment]
Sometimes this may be caused by Visual Studio TFS workspace being imported into a portable (TFC-CLC) one. The recommendation here would be to try running `tf workspaces /collection:TeamProjectCollectionUrl` as the command-line cliend suggests, e.g.

```console
tf workspaces /collection:http://myserver:8080/tfs/DefaultCollection
```

It could help with remapping the workspace.

**Comment by ForNeVeR** [comment]
While it was unlikely the original way or reproducing the issue, I've found at least some way to reproduce that.

## Stage 1: Reproduce #201 

First of all, you'll need to obtain version of the plugin where #201 was easily reproducible. For example, install Rider 2019.1.1 where version 1.151 of the Azure DevOps plugin is bundled.

1. Checkout any TFVC project containing .NET solution
2. Add an xunit test project
3. Press the **Refresh** button in the local changes list
4. Make sure **no** changes is shown: here you go, you've reproduced #201

## Stage 2: Reproduce #173

Now, even if you update the plugin to 1.152 (the latest version currently), and then go to the **Incoming** tab in the **Version control** tool window, you'll be unable to get any changes: pressing the **Update Changes** button will only show the **Messages** window with the error message:
![image](https://user-images.githubusercontent.com/92793/59674956-76a0bb80-91ee-11e9-902d-74690a2983c3.png)

The issue won't be fixed by `tf workspaces`, because it has other root cause. We should port the same fix we used for #201, but not only for `tf status` command, but for `tf get`, too.
